src/cascadia/TerminalApp/TabBase.cpp

```diff
@@ -340,6 +340,22 @@ namespace winrt::TerminalApp::implementation
         {
             _ClearTabBackgroundColor();
         }
+
+        _UpdateTabColorIndicator();
+    }
+
+    void TabBase::_UpdateTabColorIndicator()
+    {
+        const auto currentColor = GetTabColor();
+
+        if (currentColor.has_value())

```

Comment:
IIRC this if/else could probably just be a single `.value_or(Windows::UI::Colors::Transparent())`